controllers/webhook/nim_account_webhook.go

```diff
@@ -0,0 +1,118 @@
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+// +kubebuilder:webhook:admissionReviewVersions=v1,path=/validate-nim-opendatahub-io-v1-account,mutating=false,failurePolicy=fail,groups=nim.opendatahub.io,resources=accounts,verbs=create;update,versions=v1,name=validating.nim.account.odh-model-controller.opendatahub.io,sideEffects=None
+
+type nimAccountValidator struct {
+	client client.Client
+}
+
+func NewNimAccountValidator(client client.Client) admission.CustomValidator {
+	return &nimAccountValidator{client: client}
+}
+
+func (v *nimAccountValidator) ValidateCreate(ctx context.Context, obj runtime.Object) (warnings admission.Warnings, err error) {
+	log := logf.FromContext(ctx).WithName("NIMAccountValidatingWebhook")
+
+	account := obj.(*nimv1.Account)
+
+	log = log.WithValues("namespace", account.Namespace, "account", account.Name)
+	log.Info("Validating NIM Account creation")
+
+	err = v.verifySingletonInNamespace(ctx, account)
+	if err != nil {
+		log.Error(err, "Rejecting NIM Account creation because checking singleton didn't pass")
+		return nil, err
+	}
+
+	err = v.verifyAPIKeySecret(ctx, nil, account)
+	if err != nil {
+		log.Error(err, "Rejecting NIM Account creation because checking API key Secret didn't pass")
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (v *nimAccountValidator) ValidateUpdate(ctx context.Context, oldObj, newObj runtime.Object) (warnings admission.Warnings, err error) {
+	log := logf.FromContext(ctx).WithName("NIMAccountValidatingWebhook")
+
+	account := newObj.(*nimv1.Account)
+
+	log = log.WithValues("namespace", account.Namespace, "account", account.Name)
+	log.Info("Validating NIM Account update")
+
+	err = v.verifyAPIKeySecret(ctx, oldObj.(*nimv1.Account), account)
+	if err != nil {
+		log.Error(err, "Rejecting NIM Account update because checking API key Secret didn't pass")
+		return nil, err
+	}
+
+	return nil, nil
+}
+
+func (v *nimAccountValidator) ValidateDelete(ctx context.Context, obj runtime.Object) (warnings admission.Warnings, err error) {
+	// For deletion, nothing needs to be validated
+	return nil, nil
+}
+
+func (v *nimAccountValidator) verifyAPIKeySecret(ctx context.Context, oldAccount, newAccount *nimv1.Account) error {
+	if oldAccount == nil || !reflect.DeepEqual(oldAccount.Spec.APIKeySecret, newAccount.Spec.APIKeySecret) {
+		if newAccount.Spec.APIKeySecret.Name == "" {
+			msg := "name of the Secret for the API key is invalid"
+			return field.Invalid(field.NewPath("spec").Child("apiKeySecret").Child("name"), newAccount.Spec.APIKeySecret.Name, msg)
+		}
+		name := newAccount.Spec.APIKeySecret.Name
+		namespace := newAccount.Spec.APIKeySecret.Namespace
+		if namespace == "" {
```

Comment:
Yes. 
We use type `ObjectReference` for `apiKeySecret`.
In `ObjectReference`, `namespace` is an optional field.

So it is possible that users don't set `namespace`.
In this case, the value will be `""`.

If namesapce is set in the `apiKeySecret`, it will be used.
Otherwise, the code will assume the `Secret` is in the same namespace as the Account.